### PR TITLE
Rawg

### PR DESCRIPTION
**Pull request by ryanchiu2017**

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -38,10 +39,21 @@ steps:
...
     ]
+  - id: 'Deploy RAWG config'
+    dir: deploy-commands-function
+    name: golang:1.22
+    script: go run .
+    env:
+      - 'RAWG_SECRET_ID=$_RAWG_SECRET_ID'
+    secretEnv: [
+      'RAWG_SECRET_TOKEN'
+    ]
```
Comment (PinkNoize) [inline cloudbuild.yaml:50]: Remove this. We run the `deploy-commands-function` program above to register our supported discord commands: https://discord.com/developers/docs/interactions/application-commands#registering-a-command
```suggestion
```

File: cloudbuild.yaml [hunk 2]
```diff
@@ -38,10 +39,21 @@ steps:
...
 availableSecrets:
   secretManager:
   - versionName: ${_DISCORD_SECRET_ID}/versions/latest
     env: 'DISCORD_SECRET_TOKEN'
+  - versionName: ${_RAWG_SECRET_ID}/versions/latest
+    env: 'RAWG_SECRET_TOKEN'
```
Comment (PinkNoize) [inline cloudbuild.yaml:56]: The RAWG secret is not used in the cloudbuild pipeline so it is not needed
```suggestion
```

File: infra/main.tf [hunk 3]
```diff
@@ -128,6 +128,13 @@ resource "google_secret_manager_secret_iam_member" "cloud_func_member" {
   member    = "serviceAccount:${google_service_account.cloud_func_service_account.email}"
 }
 
+resource "google_secret_manager_secret_iam_member" "cloud_func_member" {
```
Comment (PinkNoize) [inline infra/main.tf:131]: Can't have 2 resources with the same name
```suggestion
resource "google_secret_manager_secret_iam_member" "cloud_func_rawg_member" {
```

File: variables.tf [hunk 4]
```diff
@@ -5,4 +5,5 @@ variable "region" {
 variable "repo_name" {}
 variable "repo_connection" {}
 variable "discord_app_id" {}
-variable "discord_public_key" {}
\ No newline at end of file
+variable "discord_public_key" {}
+variable "rawg_api_key" {}
```
Comment (PinkNoize) [inline variables.tf:9]: I'll put the api key in GCP manually after these changes have been deployed so we don't need to pass it through a variable.
As an aside, secrets passed through variables in terraform should be marked as sensitive: https://developer.hashicorp.com/terraform/tutorials/configuration-language/sensitive-variables#refactor-database-credentials
```suggestion
```

File: deploy-commands-function/go.mod [hunk 5]
```diff
@@ -2,7 +2,11 @@ module github.com/PinkNoize/flavor-of-the-week/deploy-commands-function
 
 go 1.22.7
 
-require github.com/gorilla/websocket v1.4.2 // indirect
+require (
+	github.com/dimuska139/rawg-sdk-go v1.0.1 // indirect
+	github.com/gorilla/websocket v1.4.2 // indirect
+	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
+)
```
Comment (PinkNoize) [inline deploy-commands-function/go.mod:9]: Remove the go.mod and go.sum changes unless you plan on adding changes to the go code in this PR.
`go mod tidy` should autodetect what is used and what isn't and autoremove it.

**Review by PinkNoize — CHANGES_REQUESTED** [review]
See comments.
You will also need to create the secret in main.tf and pass it to the cloudbuild trigger as a substitution. As an example, see https://github.com/PinkNoize/flavor-of-the-week/blob/23705076f7ad970ef826273d778ed5f05777d28c/main.tf#L161

**Comment by PinkNoize** [comment]
Actually, create the secret in infra/main.tf. Then you can remove all references to it in the cloudbuild.yaml files and the terraform variables

File: main.tf [hunk 6]
```diff
@@ -170,6 +177,18 @@ resource "google_secret_manager_secret" "discord_api" {
...
+
+  replication {
+    user_managed {
+      replicas {
+        location = var.region
+      }
+    }
+  }
+}
+
```
Comment (PinkNoize) [inline main.tf:191]: Move the secret to infra/main.tf

File: variables.tf (hunk 4 above)
Comment (ryanchiu2017) [reply variables.tf:9]: Do we need the same variable in infra/variables.tf?

Comment (PinkNoize) [reply variables.tf:9]: No, we can get the secret id directly from the secret resource

**Review by PinkNoize — APPROVED** [review]

**Pull request merged**